File: src/main/java/com/example/mapper/CityMapper.java
package com.example.mapper;

import com.example.domain.City;
import org.mybatis.spring.SqlSessionTemplate;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

/**
 * Created by qiuzhanghua on 16/2/24.
 */
@Component
public class CityMapper {

  @Autowired
  private SqlSessionTemplate sqlSessionTemplate;

  public City selectCityById(long id) {
    return this.sqlSessionTemplate.selectOne("selectCityById", id);
  }

}

File: src/main/kotlin/com/example/mapper/CityMapper.kt
package com.example.mapper

import com.example.domain.City
import org.mybatis.spring.SqlSessionTemplate
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component

/**
 * Created by qiuzhanghua on 16/2/24.
 */
@Component
class CityMapper {

  @Autowired
  private val sqlSessionTemplate: SqlSessionTemplate? = null

  fun selectCityById(id: Long): City {
    return this.sqlSessionTemplate!!.selectOne<City>("selectCityById", id)
  }

}